src/lib/trust_system/trust_system.ml

```diff
@@ -39,6 +39,8 @@ module Actions = struct
     | Sent_invalid_signature
         (** Peer sent us something with a signature that doesn't check *)
     | Sent_invalid_proof  (** Peer sent us a proof that does not verify. *)
+    | Sent_invalid_transition_chain_proof
```

Comment:
Can you rename this to `Sent_invalid_transition_merkle_proof` or something along those lines? I want it to be clear that `Sent_invalid_proof` refers to invalid SNARKs and this refers to invalid Merkle proofs.